### PR TITLE
Gizmos use TmpVectors and exposed materials

### LINKED ISSUE #13396: Possibility to change the gizmo colors
It could be great to access and change (Getters and Setters) the color of all the gizmos as we can already do it for the bounding box gizmo.

cf https://forum.babylonjs.com/t/change-the-color-of-the-gizmo-axes/37027

@CedricGuillemet

### PR DESCRIPTION
**Pull request by CedricGuillemet**
fixes #13396 

- Use TmpVector
- expose materials so user can tweak them

### CODE REVIEW

**Review by Popov72 — APPROVED** [review]

File: packages/dev/core/src/Gizmos/axisScaleGizmo.ts [hunk 1]
```diff
@@ -151,7 +170,7 @@ export class AxisScaleGizmo extends Gizmo implements IAxisScaleGizmo {
         this._rootMesh.addBehavior(this.dragBehavior);
 
         let currentSnapDragDistance = 0;
-        const tmpVector = new Vector3();
+
```
Comment (RaananW) [inline packages/dev/core/src/Gizmos/axisScaleGizmo.ts:173]: For future readability - you could have done this:

```javascript
const tmpVector = TmpVectors.Vector3[0];
```

and then change nothing else

**Review by RaananW — COMMENTED** [review]
LGTM! Just have a readability suggestion for the usage of temp vectors - honestly only a suggestion, this can be approved and merged.

File: packages/dev/core/src/Gizmos/axisScaleGizmo.ts (hunk 1 above)
Comment (CedricGuillemet) [reply packages/dev/core/src/Gizmos/axisScaleGizmo.ts:173]: True! I'll keep that in mind in case it happens again

Comment (RaananW) [reply packages/dev/core/src/Gizmos/axisScaleGizmo.ts:173]: feel free to resolve and merge :-)

**Pull request merged**